src/ECDSARegistryCoordinator.sol

```diff
@@ -0,0 +1,867 @@
+        // the id of the operator, which is likely the keccak256 hash of the operator's public key if using BLSRegistry
+        bytes32 operatorId;
+        // indicates whether the operator is actively registered for serving the middleware or not
+        OperatorStatus status;
+    }
+
+    /**
+     * @notice Data structure for storing operator set params for a given quorum. Specifically the 
+     * `maxOperatorCount` is the maximum number of operators that can be registered for the quorum,
+     * `kickBIPsOfOperatorStake` is the basis points of a new operator needs to have of an operator they are trying to kick from the quorum,
+     * and `kickBIPsOfTotalStake` is the basis points of the total stake of the quorum that an operator needs to be below to be kicked.
+     */ 
+     struct OperatorSetParam {
+        uint32 maxOperatorCount;
+        uint16 kickBIPsOfOperatorStake;
+        uint16 kickBIPsOfTotalStake;
+    }
+
+    /**
+     * @notice Data structure for the parameters needed to kick an operator from a quorum with number `quorumNumber`, used during registration churn.
+     * `operator` is the address of the operator to kick
+     */
+    struct OperatorKickParam {
+        uint8 quorumNumber;
+        address operator;
+    }
+
+    // TODO: doument
+    struct ECDSAPubkeyRegistrationParams {
+        address signingAddress;
+        SignatureWithSaltAndExpiry signatureAndExpiry;
+    }
+
+    /// @notice The EIP-712 typehash for the `DelegationApproval` struct used by the contract
+    bytes32 public constant OPERATOR_CHURN_APPROVAL_TYPEHASH =
+        keccak256("OperatorChurnApproval(bytes32 registeringOperatorId,OperatorKickParam[] operatorKickParams)OperatorKickParam(address operator,bytes32[] operatorIdsToSwap)");
+    /// @notice The EIP-712 typehash used for registering BLS public keys
+    bytes32 public constant PUBKEY_REGISTRATION_TYPEHASH = keccak256("ECDSAPubkeyRegistration(address operator)");
+    /// @notice The maximum value of a quorum bitmap
+    uint256 internal constant MAX_QUORUM_BITMAP = type(uint256).max;
+    /// @notice The basis point denominator
+    uint16 internal constant BIPS_DENOMINATOR = 10000;
+    /// @notice Index for flag that pauses operator registration
+    uint8 internal constant PAUSED_REGISTER_OPERATOR = 0;
+    /// @notice Index for flag that pauses operator deregistration
+    uint8 internal constant PAUSED_DEREGISTER_OPERATOR = 1;
+    /// @notice Index for flag pausing operator stake updates
+    uint8 internal constant PAUSED_UPDATE_OPERATOR = 2;
+    /// @notice The maximum number of quorums this contract supports
+    uint8 internal constant MAX_QUORUM_COUNT = 192;
+
+    /// @notice the ServiceManager for this AVS, which forwards calls onto EigenLayer's core contracts
+    IServiceManager public immutable serviceManager;
+    /// @notice the Stake Registry contract that will keep track of operators' stakes
+    IStakeRegistry public immutable stakeRegistry;
```

Comment:
ECDSAStakeRegistry does not implement IStakeRegistry. Should this be ECDSAStakeRegistry instead of IStakeRegistry? Or should ECDSAStakeRegistry implement IStakeRegistry?

Reply:
I was super liberal with removing inheritance for the sake of expediency here.
It's very possible the integration between these contracts straight-up doesn't work rn, but it would be pretty quick to check if the ECDSAStakeRegistry implements all the functions we're calling from the IStakeRegistry interface (I think it does?)
If you want this developed further, then the path forward isn't immediately clear to me.
Perhaps making a new `IECDSAStakeRegistry` interface or something would make sense, or these changes could inform _reducing_ the scope of the `IStakeRegistry` interface so both can fit into it, and then the more specific `IECDSAStakeRegistry` and `IBLSStakeRegistry` interfaces (if these need to exist at all!) can both inherit from this minimal interface.